pydatastructs/trees/binary_trees.py

```diff
@@ -415,6 +415,129 @@ def rank(self, x):
             walk = p
         return r
 
+    def _simple_path(self, key, root, path=[]):
+        """
+        Utility funtion to find the simple path between root and node.
+
+        Parameter
+        =========
+
+        key:
+            node to be searched
+
+        path:
+            Stores path from root to the node to be found
+
+        Returns
+        =======
+
+        bool:
+            True if path found else False
+        """
+        if root is None:
+            return False
+        path.append(root)
+        if self.tree[root].key == key:
+            return True
+
+        if self._simple_path(key, self.tree[root].left, path) or \
+            self._simple_path(key, self.tree[root].right, path):
+            return True
```

Comment:
Please convert this into iterative logic using `pydatastructs.Stack` as Python has a finite recursion limit irrespective of the size of heap memory.

Reply:
Stack hasn't been implemented yet.

Reply:
Yes. I made all the changes you have mentioned except this. 
pydatastruct.stack is still abstract.

Reply:
See, https://github.com/codezonediitj/pydatastructs/blob/master/pydatastructs/miscellaneous_data_structures/stack.py#L9
`Stack` can be used to remove recursion here.

Reply:
I just now saw that.
 Ill start making the changes

Reply:
Done.